Sources/AppStoreConnectCLI/Services/Operations/ReadBuildOperation.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  ReadBuildOperation.swift
+//  AppStoreConnect-Swift-SDK
+//
+//  Created by Nafisa Rahman on 1/5/20.
+//
+import AppStoreConnect_Swift_SDK
+import Combine
+import Foundation
+
+struct ReadBuildOperation {
+  private let options: ReadBuildOptions
+
+  private var listFilters: [ListBuilds.Filter]? {
+    var filters = [ListBuilds.Filter]()
+
+    if options.preReleaseVersion.isEmpty == false {
+      filters += [ListBuilds.Filter.preReleaseVersionVersion(options.preReleaseVersion)]
+    }
+
+    if options.buildNumber.isEmpty == false {
+      filters += [ListBuilds.Filter.version(options.buildNumber)]
+    }
+
+    return filters
+  }
+
+  private enum ReadBuildError: LocalizedError {
+    case noAppExist
+    case noBuildExist
+  }
+
+  init(options: ReadBuildOptions) {
+    self.options = options
+  }
+
+  func execute(with requestor: EndpointRequestor) throws -> AnyPublisher<[BuildDetailsInfo], Error> {
+      let appIds = try GetAppsOperation(
+          options: .init(bundleIds: [options.bundleId])
+      )
+      .execute(with: requestor)
+      .await()
+      .map { $0.id }
+
+      guard let appId = appIds.first else {
```

Comment:
If you look at the beta group service functions you can require an app (or an app id) in your options. And call the get app operation first in the service. I think that's the direction we are heading rather than operations calling other operations